### PR TITLE
use gometalinter with a config file

### PR DESCRIPTION
**Pull request by kdomanski**
Release-note:
```release-note
NONE
```

### CODE REVIEW

**Review by mrIncompetent — DISMISSED** [review]

File: api/Gopkg.lock [hunk 1]
```diff
@@ -1080,16 +1080,17 @@
 
 [[projects]]
   branch = "master"
-  digest = "1:3cbc05413b8aac22b1f6d4350ed696b5a83a8515a4136db8f1ec3a0aee3d76e1"
+  digest = "1:1f0e114700d8e1601436225f469f63ea9420c22f4fc01c52f9cea915eae9cdc5"
```
Comment (p0lyn0mial) [inline api/Gopkg.lock:1083]: Is there something wrong with your `dep` or with our `Gopkg.lock` file ? 

You didn't change any imports but for some reason `dep` decided to add a few dependencies.

**Review by p0lyn0mial — DISMISSED** [review]

File: api/Gopkg.lock (hunk 1 above)
Comment (p0lyn0mial) [reply api/Gopkg.lock:1083]: > bump golang.org/x/tools to a later version

oh, you decided to bump `x/tools` version

**Comment by p0lyn0mial** [comment]
before this pull `linter` step had taken 41 s to complete, after the changes 29 seconds.

File: api/Gopkg.lock (hunk 1 above)
Comment (kdomanski) [reply api/Gopkg.lock:1083]: Yeah, the old version was buggy and breaking codegen for me, see https://github.com/kubernetes/code-generator/issues/37#issuecomment-409781896

**Review by mrIncompetent — APPROVED** [review]

**Review by p0lyn0mial — APPROVED** [review]

**Pull request merged**